tests/unit/test_challenge_api.py

```diff
@@ -0,0 +1,83 @@
+import json
+import pytest
+from flask import Flask
+from io import BytesIO
+from src.core.challenge_api import api as backend_challenge_api
+
+
+@pytest.fixture
+def client():
+    app = Flask(__name__)
+    app.register_blueprint(backend_challenge_api, url_prefix="/backend")
+    client = app.test_client()
+
+    yield client
+
+
+def test_vehicle_features_post_success(client):
+    data = {
+        "vehicles": [
+            {
+                "id": "1234",
+                "features": [{"feature": "air-conditioning", "description": {"short": "AC", "long": "Air conditioning for hot days"}}],
+            }
+        ]
+    }
+
+    response = client.post("/backend/challenge", data=json.dumps(data), content_type="application/json")
+    assert response.status_code == 200
+    assert json.loads(response.data)["success"] == True
```

Comment:
As a best practice you want to use `dict.get(key, default = None)` even in tests.